src/Neo.VM/ReferenceCounter.cs

```diff
@@ -115,12 +115,12 @@ internal int CheckZeroReferred()
                             tracked_items.Remove(item);
                             if (item is CompoundType compound)
                             {
-                                references_count -= compound.SubItemsCount;
                                 foreach (StackItem subitem in compound.SubItems)
                                 {
                                     if (component.Contains(subitem)) continue;
                                     if (!NeedTrack(subitem)) continue;
                                     subitem.ObjectReferences!.Remove(compound);
+                                    references_count--;
```

Comment:
this ensures that it can be properly removed, but i think there is also problem of adding reference.

Reply:
Do you have an example?

Reply:
I believe if it goes negative it means the item was `never` met from previous one `StackItem`. You can use the `CachedCompoemts` to see.